### ISSUE TITLE
Cosmetic improvements to "junit" output format for Jenkins

### ISSUE THREAD

**Issue by jdoylei** [open-policy-agent/conftest#695]
I had two suggestions for Conftest's "junit" output format that might make it more user-friendly in Jenkins, and hopefully other CI/CD and dev tools as well.  This is with Conftest 0.30.0.

1. Group tests using the ```testcase@classname``` attribute.
2. Keep the test names more stable from build to build by basing the ```testcase@name``` attribute only on the file and namespace, and avoid including individual failure messages in the name

Regarding the first, right now every ```classname``` attribute is set to "conftest":

```
<testcase classname="conftest" name="k8s/build/k8s-app/v1_service_employee-service.yaml - pod_deny_host_path" time="0.000"></testcase>
<testcase classname="conftest" name="k8s/build/k8s-ns/networking.k8s.io_v1_networkpolicy_default-deny-all.yaml - pod_deny_host_path" time="0.000"></testcase>
```

The ```classname``` values normally help Jenkins create a nice navigable tree, when they are dot-separated Java class names (or some other dot-separated names).  But with just "conftest", all the tests show up as a very long list under a single class name "conftest" in the package name "(root)":

![image](https://user-images.githubusercontent.com/60484820/161613097-2cf8af67-4178-427a-aa75-48a6eee14d56.png)

If the ```classname``` and ```name``` were restructured like this, putting the namespace as part of ```classname``` instead of ```name```, then the Jenkins GUI could give a nice grouping by namespace:

```<testcase classname="conftest.pod_deny_host_path" name="k8s/build/k8s-ns/networking.k8s.io_v1_networkpolicy_default-deny-all.yaml" time="0.000"></testcase>```

Regarding the second, I think the failure messages were included in the name in order to keep them unique.  But maybe there's another way to note multiple failure messages, where the name could be kept stable every run (concatenating them in the ```<failure>``` element?)  Including the failure messages in ```testcase@name``` means that the name is different for a successful build and a failed build.  Like this success:

```<testcase classname="conftest" name="k8s/build/k8s-app/apps_v1_deployment_employee-service.yaml - container_deny_host_ports" time="0.000">```

vs this failure:

```<testcase classname="conftest" name="k8s/build/k8s-app/apps_v1_deployment_employee-service.yaml - container_deny_host_ports - constraint violated" time="0.000">```

This defeats the Jenkins GUI's history feature, which could be very useful with Conftest.  It makes it looks like there are two different tests - one that always succeeds and one that always fails - and they appear and disappear:

![image](https://user-images.githubusercontent.com/60484820/161615159-dca80b42-8c21-4cec-8fa3-bad20af1c643.png)
![image](https://user-images.githubusercontent.com/60484820/161615187-eab616d8-c960-4736-a92e-9b5f39cfd8f4.png)

Thank you for the great tool!

**Comment by jalseth** [comment]
Hello @jdoylei, thank you for the report. These changes seem reasonable to me, and I'd welcome a PR!

**Comment by jalseth** [comment]
@jdoylei I've implemented this in #699 and it will be included in the next release. Please review the PR and let me know if your request was accurately captured.

I opted to leave the violation message summary in the test `name` field by default as I think it provides a lot of value. For example, a K8s manifest may contain multiple containers, but only one container violates the policy; it is [common](https://github.com/plexsystems/konstraint/blob/9f91f6b88304a3bfbc8e86b55fc7da7e9cea4b98/examples/container-deny-latest-tag/src.rego#L33) to return the identifier for the violating resource and this way the test will track each container individually. If you want to hide the summary from the test name, you can use the new `--junit-hide-message` flag which was added in this change.

**Comment by jalseth** [comment]
Thanks for confirming! This will be included in the next release.
